pkg/nanotdf/nanotdf.go

```diff
@@ -0,0 +1,227 @@
+	Nanotdf_UrlProtocol__Shared Nanotdf_UrlProtocol = 255
+)
+
+type Nanotdf_EccMode uint8
+
+const (
+	Nanotdf_EccMode__Secp256r1 Nanotdf_EccMode = 0
+	Nanotdf_EccMode__Secp384r1 Nanotdf_EccMode = 1
+	Nanotdf_EccMode__Secp521r1 Nanotdf_EccMode = 2
+	Nanotdf_EccMode__Secp256k1 Nanotdf_EccMode = 3
+)
+
+type Nanotdf_CipherMode int
+
+const (
+	Nanotdf_CipherMode__Aes256gcm64Bit  Nanotdf_CipherMode = 0
+	Nanotdf_CipherMode__Aes256gcm96Bit  Nanotdf_CipherMode = 1
+	Nanotdf_CipherMode__Aes256gcm104Bit Nanotdf_CipherMode = 2
+	Nanotdf_CipherMode__Aes256gcm112Bit Nanotdf_CipherMode = 3
+	Nanotdf_CipherMode__Aes256gcm120Bit Nanotdf_CipherMode = 4
+	Nanotdf_CipherMode__Aes256gcm128Bit Nanotdf_CipherMode = 5
+)
+
+type Nanotdf_PolicyType uint8
+
+const (
+	Nanotdf_PolicyType__RemotePolicy                           = 0
+	Nanotdf_PolicyType__EmbeddedPolicyPainText                 = 1
+	Nanotdf_PolicyType__EmbeddedPolicyEncrypted                = 2
+	Nanotdf_PolicyType__EmbeddedPolicyEncryptedPolicyKeyAccess = 3
+)
+
+func deserializeBindingCfg(b byte) *Nanotdf_BindingCfg {
+	cfg := Nanotdf_BindingCfg{}
+	cfg.UseEcdsaBinding = (b >> 7 & 0x01) == 1
+	cfg.Padding = 0
+	cfg.BindingBody = Nanotdf_EccMode((b >> 4) & 0x07)
+
+	return &cfg
+}
+
+func deserializeSignatureCfg(b byte) *Nanotdf_SignatureConfig {
+	cfg := Nanotdf_SignatureConfig{}
+	cfg.HasSignature = (b >> 7 & 0x01) == 1
+	cfg.SignatureMode = Nanotdf_EccMode((b >> 4) & 0x07)
+	cfg.Cipher = Nanotdf_CipherMode(b & 0x0F)
+
+	return &cfg
+}
+
+func readPolicyBody(reader io.Reader, mode uint8) (PolicyBody, error) {
+	switch mode {
+	case 0:
+		var resourceLocator Nanotdf_ResourceLocator
+		if err := binary.Read(reader, binary.BigEndian, &resourceLocator.Protocol); err != nil {
+			return nil, err
+		}
+		if err := binary.Read(reader, binary.BigEndian, &resourceLocator.Len); err != nil {
+			return nil, err
+		}
+		body := make([]byte, resourceLocator.Len)
+		if err := binary.Read(reader, binary.BigEndian, &body); err != nil {
+			return nil, err
+		}
+		resourceLocator.Body = string(body)
+		return Nanotdf_RemotePolicy{Url: &resourceLocator}, nil
+	default:
+		var embeddedPolicy Nanotdf_EmbeddedPolicy
+		if err := binary.Read(reader, binary.BigEndian, &embeddedPolicy.Len); err != nil {
+			return nil, err
+		}
+		body := make([]byte, embeddedPolicy.Len)
+		if err := binary.Read(reader, binary.BigEndian, &body); err != nil {
+			return nil, err
+		}
+		embeddedPolicy.Body = string(body)
+		return Nanotdf_EmbeddedPolicy(embeddedPolicy), nil
+	}
+}
+
+func readEphemeralPublicKey(reader io.Reader, curve Nanotdf_EccMode) (*Nanotdf_EccKey, error) {
+	var numberOfBytes uint8
+	switch curve {
+	case Nanotdf_EccMode__Secp256r1:
+		numberOfBytes = 33
+	case Nanotdf_EccMode__Secp384r1:
+		numberOfBytes = 49
+	case Nanotdf_EccMode__Secp521r1:
+		numberOfBytes = 67
+	}
+	buffer := make([]byte, numberOfBytes)
+	if err := binary.Read(reader, binary.BigEndian, &buffer); err != nil {
+		return nil, err
```

Comment:
Generally we should be wrapping these errors, right? This should be something like 
```suggestion
		return nil, errors.Join(ErrParse, err)
```

where `ErrParse` is defined in a `const` block higher up